### ISSUE TITLE
import fails in interactive terminal

### ISSUE THREAD

**Issue by aragong** [microsoft/vscode-jupyter#4251]
Issue Type: <b>Bug</b>

When I run my python project in a standard terminal it works properly. However, If I run the same project in the interactive terminal fails and raise this error when it tries to load my codes in folder "codes":

> import imp...
>---------------------------------------------------------------------------
>ModuleNotFoundError                       Traceback (most recent call last)
>d:\2020_SICMA\SICMA_MODULO_OPERACIONAL\Codigo_operacional\main.py in ()
>      2 import json
>      3 import os
>----> 4 from code.data_repository import DataRepository
>      5 from code.environment import Environment
>      6 

ModuleNotFoundError: No module named 'code.data_repository'; 'code' is not a package

I also add a screenshot of my folder structure and the main.py script where the flow breaks. Thank you in advance!
![image](https://user-images.githubusercontent.com/48764870/102588183-5288a800-410d-11eb-81bc-c0822c23c4b3.png)

 ```import imp
import json
import os
from code.data_repository import DataRepository
from code.environment import Environment

import pandas as pd

# Workarround for logins in .key file that is excluded in .gitignore
key = imp.load_source(os.getcwd(), "authentifications.key")


# NOTE - INFO provided by the controller
# -----------------------------------
data_controller = {
    "project": "SICMA",
    "refineryId": 2,
    "model": "Teseo",
    "spill_point_X": [-6.925839],
    "spill_point_Y": [37.179055],
    "substanceIds": [7],
    "mass": [400],
}
json_str = json.dumps(data_controller)
# -----------------------------------

# Load Json (this object will be created bu OperationalController or WebApi)
controller_info = json.loads(json_str)


def run_spill_simulation():

    # TODO - Set a Try/catch

    # 1. REQUEST INFO FROM DATABASES
    info = DataRepository(url=key.SICMA_api_url, auth=key.SICMA_api_auth)
    refinery = info.get_refinery(refineryId=controller_info["refineryId"])
    providers = info.get_providers(refineryId=controller_info["refineryId"])
    substances = info.get_substances(substanceIds=controller_info["substanceIds"])


    # 2. INIT ENVIRONMENT
    SICMA = Environment(project=controller_info["project"])
    SICMA.clean_jobs()


    # 3. CREATE NEW JOB
    job = SICMA.create_job(model=controller_info["model"], location=refinery["name"])


    # 4. GET FORCINGS
    forcings = []
    for provider in providers:
        forcings.append(provider["forcingName"])
        SICMA.get_forcing4Teseo(job=job, provider=provider, auth=key.datahub)
    # FIXME - Will disapeard, the model shouldn't require files for forcings that will not used!!
    SICMA.create_null_Forcings4Teseo(job=job, forcings=forcings)


    # 5.CONFIGURE JOB
    SICMA.configure_job(job=job, controller_info=controller_info, substances=substances)


    # 6. EXECUTE JOB
    SICMA.run_job(job=job)
    print("Simulation finished!")

    # 8. POSTPROCESS RESULTS
    SICMA.postprocess_results(job=job)


    # TODO - Uncomment
    # return job


def archive_job(job):
    SICMA = Environment(project=controller_info["project"])
    SICMA.archive_job(job=job)


def clean_archive_and_database():
    pass


if __name__ == "__main__":
    run_spill_simulation()
```

Extension version: 2020.12.414227025
VS Code version: Code - Insiders 1.53.0-insider (c927a8015b9e26bd454d6e293bb0384aa1975d06, 2020-12-18T05:27:48.567Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz (8 x 3411)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.98GB (9.74GB free)|
|Process Argv|--crash-reporter-id dcc88fd9-fed3-4050-a9e9-4928cc13bd3d|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
openlogontheside:30221882
python763:30178808
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
python504:30227505
vswsl492:30208929
wsl2prompt:30219162
vstry244:30230484
pythonvsdeb440:30224570
unusedpromptcf:30219165
folderexplorercf:30219167
openfilemenucf:30219169
pythonvsded773:30223139
core-portspanel:30233467

```

</details>

**Comment by claudiaregio** [comment]
Hi @aragong, can you please provide the following information? 

-   VS Code version: XXX
-   Jupyter Extension version (available under the Extensions sidebar): XXX
-   Python Extension version (available under the Extensions sidebar): XXX
-   OS (Windows | Mac | Linux distro) and version: XXX
-   Python and/or Anaconda version: XXX
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): XXX
-   Jupyter server running: Local | Remote | N/A

Can you also please provide a screenshot of the entire screen when the error occurs or a screen recording (https://www.screentogif.com/) of you running into the error? (Please make sure to include the entire screen so we don't miss any important information)

**Comment by joyceerhl** [comment]
Hi @aragong, thanks for filing a bug. Could I ask you to try running the following code in the interactive window and let us know what the output is?

```python
import os
print(os.getcwd())
```

My guess is that you're running into https://github.com/microsoft/vscode-jupyter/issues/1241. In other words the interactive window's current working directory is not the same as the terminal's working directory, so relative file imports are incorrectly resolved.

**Comment by aragong** [comment]
> Hi @aragong, thanks for filing a bug. Could I ask you to try running the following code in the interactive window and let us know what the output is?
> 
> ```python
> import os
> print(os.getcwd())
> ```
> 
> My guess is that you're running into #1241. In other words the interactive window's current working directory is not the same as the terminal's working directory, so relative file imports are incorrectly resolved.

thank you for your response, it seems that the rootpath is ok! see the image bellow with the os.getcwd():
![image](https://user-images.githubusercontent.com/48764870/102751812-c7a3e980-4368-11eb-9e5f-bd86a2deed61.png)

**Comment by aragong** [comment]
> Hi @aragong, can you please provide the following information?
> 
> * VS Code version: XXX
> * Jupyter Extension version (available under the Extensions sidebar): XXX
> * Python Extension version (available under the Extensions sidebar): XXX
> * OS (Windows | Mac | Linux distro) and version: XXX
> * Python and/or Anaconda version: XXX
> * Type of virtual environment used (N/A | venv | virtualenv | conda | ...): XXX
> * Jupyter server running: Local | Remote | N/A
> 
> Can you also please provide a screenshot of the entire screen when the error occurs or a screen recording (https://www.screentogif.com/) of you running into the error? (Please make sure to include the entire screen so we don't miss any important information)

thank you for your response, I post the information you request bellow:
Version: 1.53.0-insider (user setup)
Commit: c927a8015b9e26bd454d6e293bb0384aa1975d06
Date: 2020-12-18T05:27:48.567Z
Electron: 11.0.5
Chrome: 87.0.4280.88
Node.js: 12.18.3
V8: 8.7.220.29-electron.0
OS: Windows_NT x64 10.0.18363

Python 3.7.8 64bit

jupyter_client            6.1.7                      py_0    conda-forge
jupyter_core              4.6.3            py37hc8dfbb8_2    conda-forge
jupyterlab                2.2.8                      py_0    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_server         1.2.0                      py_0    conda-forge

ipython                   5.8.0                    py37_1    conda-forge

**Comment by joyceerhl** [comment]
Thanks @aragong, I can repro this after setting up the following directory structure and creating the IW with the same cwd as the parent directory of `code`:

|
|--code
       |--__init__.py
       |--main.py

This is apparently [also a problem](https://medium.com/@dustinmichels/three-handy-jupyter-notebook-snippets-dcb2f8ddc7db) in browser based notebooks. See also https://stackoverflow.com/questions/43120112/module-not-found-during-import-in-jupyter-notebook.

**Comment by aragong** [comment]
thank you @joyceerhl,

So I am a little bit confused... Is it something to be fixed? or I have to follow the workarounds to solve the problem?

Thank you in advance!

**Comment by aragong** [comment]
I found an important thing @joyceerhl! If I refactor the name of the package "code" for any other like "teseo_model" imports works properly in both, terminal and IW. 

So I realize that it is any problem because it already exist a package named "code" I do not know if this behaviour is considered a bug to be fixed or not. But if I used the word "code" for this folder, it seems that in the IW loads the default code package instead of the one I created. This is very confusing, it will be great if the behaviour will be the same in the terminal and in the IW.

I hope this will be of some help to homogenize the behaviour between terminal and IW!
